drivers/ethernet/eth_xlnx_gem.c

```diff
@@ -321,7 +316,7 @@ static void eth_xlnx_gem_isr(const struct device *dev)
 	 * cleared whenever the corresponding work item submitted from within
 	 * this ISR is being processed.
 	 */
-	sys_write32((0xFFFFFFFF & ~(ETH_XLNX_GEM_IXR_FRAME_RX_BIT |
+	sys_write32((~0 & ~(ETH_XLNX_GEM_IXR_FRAME_RX_BIT |
```

Comment:
why is this 32-bit masking necessary? it already calls `sys_write32`, which cannot write more than 32-bits anyway.

also if any, it should be:
```suggestion
	sys_write32((~0U & ~(ETH_XLNX_GEM_IXR_FRAME_RX_BIT |
```